decavision/dataset_preparation/generate_pseudolabels.py

```diff
@@ -0,0 +1,216 @@
+import ast
+import os
+import shutil
+from distutils.dir_util import copy_tree
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from tensorflow.keras.models import load_model
+from tqdm import tqdm
+
+
+class PseudoLabelGenerator:
+    """
+    Class to generate pseudo labels.
+    Arguments:
+        model_path (str): path to trained model
+        train_path (str): path to training data
+        unlabeled_path (str): path to unlabeled data
+        pseudo_path (str): path to store train data and pseudo data
+    """
+
+    def __init__(self, model_path, train_data_path, unlabeled_path, pseudo_data_path):
+        self.model_path = model_path
+        self.train_data_path = train_data_path
+        self.unlabeled_path = unlabeled_path
+        self.pseudo_data_path = pseudo_data_path
+
+        # Load model
+        model = load_model(self.model_path, compile=False)
+        self.model = model
```

Comment:
You can directly set self.model to be load.model.

Reply:
Done.